### PR TITLE
Drop old capture loading.

### PR DESCRIPTION
**Pull request by florian-kuebler**
Old captures are not supported due to the changes on manual
instrumentation.

So let's drop all related code.

Bug: http://b/196374661
Test: Try to load an old and a new capture.

### CODE REVIEW

**Comment by florian-kuebler** [comment]
> Can you clarify what "old" here means? What support are we breaking and what exactly does that mean for end-users? I thought we were supporting both the old and new manual instrumentation for a while.

The format that is based on the client protos in `capture_data.proto`. The new format is the one based on `CaptureFile` and contains the proto messages that come from the service (`capture.proto`).

**Comment by florian-kuebler** [comment]
We dropped support for writing in this format a couple of releases ago, but kept support for reading, basically as it was still possible.

File: src/OrbitGl/App.cpp [hunk 1]
```diff
@@ -1189,9 +1188,8 @@ Future<ErrorMessageOr<CaptureListener::CaptureOutcome>> OrbitApp::LoadCaptureFro
     if (capture_file_or_error.has_value()) {
       load_result = LoadCaptureFromNewFormat(this, capture_file_or_error.value().get(),
                                              &capture_loading_cancellation_requested_);
-    } else {  // Fall back to old capture format.
-      load_result = orbit_client_model::capture_deserializer::Load(
-          file_path, this, module_manager_.get(), &capture_loading_cancellation_requested_);
+    } else {  // The old format is not supported anymore.
+      load_result = ErrorMessage("Old captures are not supported anymore.");
```
Comment (dimitry-) [inline src/OrbitGl/App.cpp:1192]: We should probably use capture_file_or_error.error() here because it provides more context.

Comment (florian-kuebler) [reply src/OrbitGl/App.cpp]: Makes sense. Done!

**Comment by ronaldfw** [comment]
> We dropped support for writing in this format a couple of releases ago, but kept support for reading, basically as it was still possible.

Ok, so this is the very first, somewhat slow version of manual instrumentation. Then I don't have concerns with removing this.

**Comment by florian-kuebler** [comment]
> > We dropped support for writing in this format a couple of releases ago, but kept support for reading, basically as it was still possible.
> 
> Ok, so this is the very first, somewhat slow version of manual instrumentation. Then I don't have concerns with removing this.

You mean, the very first version of serialization? The change itself has nothing to do with manual instrumentation. It was only triggered by the fact, that we now dropped support for reading the old manual instrumentation events.

File: src/OrbitGl/App.cpp [hunk 2]
```diff
@@ -1189,9 +1188,8 @@ Future<ErrorMessageOr<CaptureListener::CaptureOutcome>> OrbitApp::LoadCaptureFro
     if (capture_file_or_error.has_value()) {
       load_result = LoadCaptureFromNewFormat(this, capture_file_or_error.value().get(),
                                              &capture_loading_cancellation_requested_);
-    } else {  // Fall back to old capture format.
-      load_result = orbit_client_model::capture_deserializer::Load(
-          file_path, this, module_manager_.get(), &capture_loading_cancellation_requested_);
+    } else {  // The old format is not supported anymore.
```
Comment (dimitry-) [inline src/OrbitGl/App.cpp:1191]: nit: remove this comment? the error might be about not just the old format. could be an invalid capture file.

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

**Pull request merged**